CONTRIBUTING.md

```diff
@@ -72,6 +73,30 @@ pip install -r requirements-dev.txt
 pip install -e .  # installs koalas from current checkout
 ```
 
+## Working with Git
+
+[The instruction on Github](https://help.github.com/set-up-git-redirect) has all steps you need to take to install git, set up
```

Comment:
Hm, my feeling is that this guide is too general but not specific to Koalas.
For instance, I think we should clarify how to rebase or merge upstream to resolve conflicts.

@RainFung, do you have some ideas on this? what should we write for `git` beginners?

Reply:
The reason to add it is: When I start contributing to new open source project, I will first of all read `CONTRIBUTING.md`, and in a normal flow, I think once people get environment set up, and next thing should be working with Git, e.g. forking and cloning Koalas. And then this is just the simplest example to present how to create a branch on Koalas and work on their implemented features instead of directly on Koalas master. Regarding how to rebase/merge, I believe people should find it on a github website or learn it themselves.

But this is just my thought when reading through the `CONTIRBUTING.md` and feeling something might be missing/nice to add. If you think this is unrelated to Koalas, feel free to close this PR @HyukjinKwon

Reply:
Let's leave this one closed for now.